### ISSUE TITLE
Add documentation to the Stax filters feature

### ISSUE THREAD

**Issue by brandelune** [omegat-org/omegat#421]
https://github.com/omegat-org/omegat/pull/356

The feature relates to the file filters and to file formats.
The following files need to be modified:

https://github.com/omegat-org/omegat/blob/master/doc_src/en/HowTo_TranslateOtherFiles.xml
https://github.com/omegat-org/omegat/blob/master/doc_src/en/App_FileFilters.xml

**Comment by brandelune** [comment]
https://github.com/omegat-org/omegat/pull/356#issuecomment-1382692284 @t-cordonnier one of the filter seems to be for SDL packages. Does that mean that the plugin creates a valid return package ? I could not find that information here: http://185.13.37.79/?q=/node/88

**Comment by brandelune** [comment]
The MSO filter has the exact same options as OmegaT's default with a very similar name.

Now that the Stax filter is integrated at the same level as OmegaT's (or is it?) we need to clarify the naming and give an explanation about why we keep the two.

Are we slowly phasing out the original one ? Do we suggest new users to use yours instead ? And Why ?

That's the kind of info I'd like to summarize in the Filter explanation.

**Comment by t-cordonnier** [comment]
> Does that mean that the plugin creates a valid return package ? 

This filter reads and updates the SDLXLIFF files contained in the package, for that reason it returns a valid _input_ package (sdlppx) which you can use in Trados to continue the translation.
To create a return package (sdlrpx) I think it should also do some changes in the project file (sdlproj) which actually it does not. This is something I could consider as an option for the future.

That said this filter was mostly an experiment to check feasibility of zip-based formats, and build the ZIP architecture which I finally used for OpenXML. Now you should test it to say whenever it is really useful or not: maybe other approach similar to MED and OMT packages may be preferable in the future.

In particular, this filter does not yet make anything from the translation memories (SDLTM) which are in the package. A real package approach, instead of seeing it as a file format, may be better.

Test and tell me what you think.

**Comment by t-cordonnier** [comment]
> Are we slowly phasing out the original one ? Do we suggest new users to use yours instead ? And Why ?

Personally I may say yes and consider that since the new filter has more features than the old one, the only reason why to keep the old one is to preserve compatibility, as a translation made with old filter cannot be fully compatible with the new one (first of all because the tags are different)
But it is up to you to decide.

**Comment by brandelune** [comment]
> Test and tell me what you think.

I don't work with SDL packages so I would not be able to test. Just FYI Swordfish offers the return package + SDLTM support. Most LSP would expect the translator to have used the internal TM and to deliver a return package, so, to make sure that users are not confused, maybe it would be better to add _(test)_ after its name then.

**Comment by brandelune** [comment]
> > Are we slowly phasing out the original one ? Do we suggest new users to use yours instead ? And Why ?
> 
> Personally I may say yes [to phasing out] and consider that since the new filter has more features than the old one, the only reason why to keep the old one is to preserve compatibility, as a translation made with old filter cannot be fully compatible with the new one (first of all because the tags are different) But it is up to you to decide.

Ok. So we need to use the same names but with some extension. I don't think "stax" means anything to users, so we could use a _(legacy)_ and a _(new)_ suffix. @miurahr @kosivantsov @Kazephil what do you think?

And we'd need to add info related to the difference between the two.

Do you have a "conversion" script or something to ease the transition between the two ?

**Comment by t-cordonnier** [comment]
> And we'd need to add info related to the difference between the two.

Main differences here : http://185.13.37.79/?q=node/133

> Do you have a "conversion" script or something to ease the transition between the two ?

If it were possible I would not have said that we should keep the old one for compatibility...

That said, the problem of compatibility only concerns translations which were previously made with the old filter. If I started a new translation now I would see no reason why to start it with the old filter. But I know that some of your users are very conservative.

**Comment by brandelune** [comment]
> > And we'd need to add info related to the difference between the two.
> 
> Main differences here : http://185.13.37.79/?q=node/133

Thank you.

> > Do you have a "conversion" script or something to ease the transition between the two ?
> 
> If it were possible I would not have said that we should keep the old one for compatibility...

:-)

> That said, the problem of compatibility only concerns translations which were previously made with the old filter. If I started a new translation now I would see no reason why to start it with the old filter. But I know that some of your users are very conservative.

Well, starting with 5.8 they're going to be your users too... ;-)

Documentation + the suffix should be enough to remove all possible confusion.

**Comment by Kazephil** [comment]
> Ok. So we need to use the same names but with some extension. I don't think "stax" means anything to users, so we could use a _(legacy)_ and a _(new)_ suffix. @miurahr @kosivantsov @Kazephil what do you think?

If the new StaX filter does everything the old one does while offering new features, I agree that the old filter should be phased out.

Mayne something along the lines of:
- XLIFF_Input
  This filter supersedes the legacy filter and should be user for new projects.
- XLIFF_Legacy
  This filter bas been superseded by the newer input filter and should only be used for projects already created with this filter.

Note: I picked "_Input" somewhat arbitrarily based on Thomas's comment that it creates an input package.

Other possible suffixes:
- _Experimental
- _Test
- _New
- _Advanced

The first two might sound scary and unstable to users, however!

(Excuse any weird typos or formatting. I'm responding on mobile.)

**Comment by miurahr** [comment]
https://github.com/omegat-org/omegat/blob/master/src/org/omegat/Bundle.properties#L1509-L1516

```
# org.omegat.filters4.xml.xliff.Xliff1Filter.java
XLIFF1FILTER_FILTER_NAME=XLIFF 1.x Bilingual

# org.omegat.filters4.xml.xliff.Xliff2Filter.java
XLIFF2FILTER_FILTER_NAME=XLIFF 2.x Bilingual

```

Filter name is now "XLIFF 1.x Bilingual".

**Comment by brandelune** [comment]
XLIFF 1.2 was released 14 years ago. We don't have any XLIFF v.1 after that 1.2, so I propose that we use XLIFF 1.2, or even XLIFF v1.

Same for XLIFF 2.x, unless there are profound modifications to the standard that require 2.2 to have support different from 2.1 or 2.0, I don't see why we can't call the filter in a way that more readable, like XLIFF v2.

Also, by definition, XLIFF is a bilingual format. So I'd rather have the Stax filters be labeled the most normal way possible and the original OmegaT filters be called "legacy" or something.

So we'd have:

XLIFF → XLIFF (legacy)
Microsoft Open XML → Microsoft Office Open XML (legacy)
XLIFF 1.x → XLIFF v1
XLIFF 2.x → XLIFF v2
SDL project → SDL project (beta)
SDL XLIFF → same name
Microsoft Office Open XML → same name
MemoQ XLIFF → memoQ XLIFF (official tool name)

Eventually, the SDL filters will have to be renamed RWS...

What do you all think?

**Comment by Kazephil** [comment]
Maybe the SDL ones can simply be named "Trados", since the name of the product seems less likely to change than the name of the company that owns it. :laughing: 

One thing, though: with that naming scheme, "XLIFF (legacy)" sounds like a pre-XLIFF v.1 version of the format rather than a legacy version of the filter. If it's the filter rather than the format that's "legacy" (the impression I had), the name should probably be "XLIFF (legacy filter)" or "XLIFF (legacy monolingual filter)".

Other than that, your proposed list looks fine to me (with the caveat that I don't work directly with XLIFF and am not very familiar with its nuances).

**Comment by brandelune** [comment]
> Maybe the SDL ones can simply be named "Trados", since the name of the product seems less likely to change than the name of the company that owns it. 😆

If ever RWS decides to use rwsxliff/rwsppx as file extensions, I guess we'll be able to adapt the filter then. Let's worry about that when it happens 😉

> One thing, though: with that naming scheme, "XLIFF (legacy)" sounds like a pre-XLIFF v.1 version of the format rather than a legacy version of the filter.

The thing is that most people who don't care and want a _normal_ behavior will have the new filter enabled by default. Only people who **want** the old behavior need to be aware of the meaning, and that is a documentation issue (something we'll put in the manual and _also_ in the release announcement and eventually in the changes.txt file too).

> Other than that, your proposed list looks fine to me (with the caveat that I don't work directly with XLIFF and am not very familiar with its nuances).

Mostly proprietary extensions that are accepted in the standard and that allow software vendors to make their XLIFF incompatible with other vendor's tools 😁.

**Comment by t-cordonnier** [comment]
> Eventually, the SDL filters will have to be renamed RWS...
> Maybe the SDL ones can simply be named "Trados", since the name of the product seems less likely to change than the name of the company that owns it. laughing

The extension of the file formats is still sdlxliff and sdlppx, despite the change of the name of the company. And I am pretty sure lot of people are not aware of this change so they will still see the file extension. Personally for this reason I would keep sdlxliff and sdl project.

**Comment by brandelune** [comment]
Thank you both for your comments.

What about this:

XLIFF → XLIFF (legacy filter)
Microsoft Open XML → Microsoft Office Open XML (legacy filter)
XLIFF 1.x → XLIFF (like Okapi names it, but without the Okapi suffix, that way we know it's "ours")
XLIFF 2.x → XLIFF 2 (like Okapi names it, but without the Okapi suffix, that way we know it's "ours")
SDL project → SDL project (beta)
SDL XLIFF → same name
Microsoft Office Open XML → same name
MemoQ XLIFF → memoQ XLIFF (official tool name)

**Comment by brandelune** [comment]
> #356
> 
> The feature relates to the file filters and to file formats. The following files need to be modified:
> 
> https://github.com/omegat-org/omegat/blob/master/doc_src/en/HowTo_TranslateOtherFiles.xml https://github.com/omegat-org/omegat/blob/master/doc_src/en/App_FileFilters.xml

We also need to change the bundle.

**Comment by brandelune** [comment]
https://github.com/omegat-org/omegat/pull/444

**Comment by brandelune** [comment]
The memoQ XLIFF filter does not appear anymore? Is it intended?

**Comment by t-cordonnier** [comment]
MemoQ filter is highly experimental, I even have the case where it broke the full plugin. And it does not yet include lot of features from MemoQ, and was not tested, mainly because I did not have lot of samples.
I did not include it in the patches I made for you.

In DGT / plugin web site I had multiple versions of the plugin and only the last one (2.1) contains it:
http://185.13.37.79/sites/default/files/upload/stax-filters/files/
The one which is included in OmegaT is the 2.0, which I consider more stable.

We can consider it for later... if we find a way to test it now that the other variants are in the core of OmegaT.
